jupyter_console/ptshell.py

```diff
@@ -708,22 +708,22 @@ def handle_iopub(self, msg_id=''):
     def handle_rich_data(self, data):
         for mime in self.mime_preference:
             if mime in data and mime in self._imagemime:
-                self.handle_image(data, mime)
-                return True
+                return self.handle_image(data, mime)
```

Comment:
we don’t want to return `False` if one mime can’t be handled. maybe the next can?